codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketServerProtocolHandler.java

```diff
@@ -57,10 +59,35 @@
     public enum ServerHandshakeStateEvent {
         /**
          * The Handshake was complete succesful and so the channel was upgraded to websockets
+         *
+         * @deprecated in favor of {@link HandshakeComplete} class,
+         * it provides extra information about the handshake
          */
+        @Deprecated
         HANDSHAKE_COMPLETE
     }
 
+    /**
+     * Marker interface for events that are fired to notify about handshake status
+     */
+    public interface ServerHandshakeEvent {  }
```

Comment:
why we even need this and not just use HandshakeComplete event directly as we need to cast anyway if we want to do anything.